cms/bundles/viewsets/bundle.py

```diff
@@ -464,19 +466,130 @@ def get_teams_display_value(self) -> str:
         value: str = self.object.get_teams_display()
         return value
 
-    def get_bundled_datasets_display_value(self) -> str:
-        """Returns formatted markup for datasets linked to the Bundle."""
-        if self.object.bundled_datasets.exists():
-            datasets = Dataset.objects.filter(pk__in=self.object.bundled_datasets.values_list("dataset__pk", flat=True))
-            return format_html(
-                "<ol>{}</ol>",
-                format_html_join(
-                    "\n",
-                    '<li><a href="{}" target="_blank" rel="noopener">{}</a></li>',
-                    ((bundled_dataset.website_url, bundled_dataset) for bundled_dataset in datasets),
-                ),
+    @staticmethod
+    def get_human_readable_state(state: str) -> str:
+        """Converts a machine-readable state string to a human-readable format."""
+        if not state or state == MISSING_VALUE:
+            return MISSING_VALUE
+        match state:
+            case "APPROVED":
+                return "Approved"
+            case "PUBLISHED":
+                return "Published"
+            case _:
+                return state.replace("_", " ").title()
+
+    def _get_processed_datasets(self) -> list[dict[str, "SafeString | str"]]:
+        """Fetches and processes dataset information from the bundle API."""
+        client = BundleAPIClient(access_token=self.request.COOKIES.get(settings.ACCESS_TOKEN_COOKIE_NAME))
+        bundle_contents = client.get_bundle_contents(self.object.bundle_api_content_id)
+        processed_data = []
+
+        for content_item in bundle_contents.get("contents", []):
+            if content_item.get("content_type") == "DATASET":
```

Comment:
Nit: It would be good to invert this to return early, which would consequently reduce the indentation and improve readability.